src/io/rme_result_io.jl

```diff
@@ -0,0 +1,279 @@
+
+    # Counterfactual scenario if outplant area and enrichment area are both 0
+    scenario_groups::Dict{Symbol, BitVector} =
+        _construct_scenario_groups(inputs, length(raw_set.scenarios))
+
+    env_layer_md::EnvLayer = EnvLayer(
+        data_dir,
+        geodata_path,
+        reef_id_col,
+        cluster_id_col,
+        "",
+        joinpath(data_dir, "con_bin"),
+        "",
+        "",
+        timeframe
+    )
+
+    outcomes::Dict{Symbol, YAXArray} = Dict();
+    for (key, cube) in raw_set.cubes
+        outcomes[key] = _reformat_cube(RMEResultSet, cube)
+    end
+
+    # Calculate relative cover
+    n_locations::Int = length(raw_set.locations)
+    outcomes[:relative_cover] =
+        (outcomes[:total_cover] ./ 100) ./ reshape(geodata.k, (1, n_locations, 1))
+    outcomes[:relative_taxa_cover] =
+        (outcomes[:total_taxa_cover] ./ 100) ./ reshape(geodata.k, (1, n_locations, 1))
+
+    mod_spec::DataFrame = _create_model_spec(inputs)
+
+    return RMEResultSet(
+        name,
+        netcdf_rcp,
+        site_ids,
+        geodata.area,
+        location_max_coral_cover,
+        location_centroids,
+        env_layer_md,
+        connectivity,
+        geodata,
+        scenario_groups,
+        inputs,
+        SimConstants(),
+        mod_spec,
+        outcomes
+    )
+end
+
+"""
+    _get_inputs(filepath::String)
+
+Read in scenario dataframe from given location. Warn and default to empty dataframe if not
+found.
+"""
+function _get_inputs(filepath::String)::DataFrame
+    if !isfile(filepath)
+        @warn "Unable to find scenario spec at $(filepath). Skipping."
+        return DataFrame()
+    end
+    return inputs = CSV.read(filepath, DataFrame, header=true)
+end
+
+"""
+    _construct_scenario_groups(inputs::DataFrame, n_scenarios::Int)::Dict{Symbol, BitVector}
+
+Using the inputs dataframe to determine if a run was counterfactual, construct the scenario
+groupings for compatibility with ADRIA scenario plotting.
+"""
+function _construct_scenario_groups(
+    inputs::DataFrame,
+    n_scenarios::Int
+)::Dict{Symbol, BitVector}
+    if size(inputs) == (0, 0)
+        return Dict(:counterfactual => BitVector([true for _ in 1:n_scenarios]))
+    end
+    counterfactual_scens::BitVector = BitVector([
+        p_a == 0.0 && e_a == 0 for (p_a, e_a)
+            in zip(inputs.outplant_area_pct, inputs.enrichment_area_pct)
+    ])
+    # Intervened if not counterfactual
+    intervened_scens::BitVector = BitVector([
+        p_a != 0 || e_a != 0 for (p_a, e_a)
+            in zip(inputs.outplant_area_pct, inputs.enrichment_area_pct)
+    ])
+    scenario_groups::Dict{Symbol, BitVector} = Dict()
+
+    # If the runs do not contain counterfactual or intervened runs exclude the key
+    if any(counterfactual_scens)
+        scenario_groups[:counterfactual] = counterfactual_scens
+    end
+    if any(intervened_scens)
+        scenario_groups[:intervened] = intervened_scens
+    end
+    return scenario_groups
```

Comment:
Same here: spacing for readability.

Note `return` statements, with very few exceptions, should always be separated for clarity.

```suggestion
    # If the runs do not contain counterfactual or intervened runs exclude the key
    if any(counterfactual_scens)
        scenario_groups[:counterfactual] = counterfactual_scens
    end

    if any(intervened_scens)
        scenario_groups[:intervened] = intervened_scens
    end

    return scenario_groups
```